Random Vogon fixes

vanam opened the pull request:
Hi,
I tried your guide and fixed several small issues:

* I made import paths consistent 395d20517a1230830952250c40893ea3a089457d
* I used lazy Gradle API 7d3e303f2757cd4c97d9692a3ce2a2261c435ff2
* My IDE (IDEA) was unable to find generated classes, so I added them to source set 9970af7106db43b93f6804bcfae3ebc9a36d1ad8
* I bumped libraries to their latest versions 9970af7106db43b93f6804bcfae3ebc9a36d1ad8
* I added `jooqCodegen` to be called before java compile. I want everything to be done by `./gradlew build`, however; I would understand if you didn't like this change. c0b87762faaa78044171c80c9aa0ade19b2bc3b7
* Use Gradle in GitHub Actions. ec0b6ed0780f6318d61f6bd8dd6e52336e87b4b6

So Long, and Thanks for All the Fish

vanam committed:
Fix static import paths.

vanam committed:
Use lazy gradle API.

vanam committed:
Bump dependency versions.

vanam committed:
Call jooq coodegen before java compile.

vanam committed:
Use Gradle in GitHub Actions.

sergeylappo commented:
Thanks a lot for the valuable contribution, and sorry for the long delay.

> I made import paths consistent https://github.com/sergeylappo/tc-guide-working-with-jooq-flyway-using-testcontainers-gradle/commit/395d20517a1230830952250c40893ea3a089457d

Nice one.

> My IDE (IDEA) was unable to find generated classes, so I added them to source set

This issue is known to me, and I believe it's a jooq plugin bug. 
This was broken in Jooq 3.19.6 and seems to be fixed in Jooq 3.19.10. Please check again.

> I added jooqCodegen to be called before java compile. I want everything to be done by ./gradlew build, however; I would understand if you didn't like this change.

Nice one. The whole idea of the guide was just do `./gradlew build`, thus I would love to incorporate this.
The "issue" appears to be new, and which is introduced with the newer plugin versions (see [here](https://github.com/jOOQ/jOOQ/commit/902a5c9d9fb93fc95259a585417c3e1322d6ebda#diff-fd1cc29a1e96274bf4cf4ad91fd38b0b43aa7dea221128973b7a8da6e88915fdL94), 3.19.2)



> Use Gradle in GitHub Actions. 

👍

sergeylappo commented on build.gradle:
```diff
@@ -39,6 +39,15 @@ version = '1.0.0-SNAPSHOT'
 description = 'tc-guide-working-with-jooq-flyway-using-testcontainers-gradle'
 java.sourceCompatibility = 17
 
+def jooqGeneratedDir = layout.getBuildDirectory().dir("generated-sources/jooq").get().asFile.path
+sourceSets {
+    main {
+        java {
+            srcDirs += [jooqGeneratedDir]
+        }
+    }
+}
+
```
This can be removed as of jooq 3.19.10
```suggestion
```

sergeylappo commented on settings.gradle:
```diff
@@ -7,11 +7,11 @@ rootProject.name = 'tc-guide-working-with-jooq-flyway-using-testcontainers-gradl
-            version('postgresql', '42.7.1')
+            version('flyway', '10.13.0')
+            version('testcontainers', '1.19.8')
+            version('jooq', '3.19.8')
```
```suggestion
            version('jooq', '3.19.10')
```

sergeylappo commented on build.gradle:
```diff
@@ -138,7 +148,7 @@ jooq {
             target {
                 packageName = 'com.testcontainers.demo.jooq'
-                directory = 'target/generated-sources/jooq'
+                directory = jooqGeneratedDir
```
Would need to be reverted as well

sergeylappo commented:
Would be happy to merge once the comments are incorporated.
Thanks again for the contribution, also mentioned you [here](https://sergeylappo.github.io/posts/gradle-jooq-flyway-testcontainers/#enforce-execution).

vanam commented:
Comments incorporated

vanam committed:
Use TOML as version catalog.

vanam committed:
Use Java 21 toolchain.

vanam commented:
I added two more things:

1) I wondered why there are two places to define library versions (`libs.versions.toml` and `settings.gradle`). I read the[ documentation](https://docs.gradle.org/current/userguide/platforms.html#sub::toml-dependencies-format) and ended up using TOML file

2) Instead of using `java.sourceCompatibility` I resorted to using `java.toolchain` and updated Java version to latest LTS version - 21.

vanam committed:
Use mavenCentral() instead of explicit URL.

vanam committed:
Remove redundant mavenCentral().

sergeylappo commented:
Although I'm not convinced about [toolchains yet](https://jakewharton.com/gradle-toolchains-are-rarely-a-good-idea/) those are very good improvements.

sergeylappo commented:
Thanks a lot for the contribution, and I'm sorry for the long review time.
